src/app/shared/nav/nav.component.ts

```diff
@@ -1,16 +1,40 @@
-import { Component } from '@angular/core';
-import { RouterLink, RouterLinkActive } from '@angular/router';
+import { Component, OnDestroy } from '@angular/core';
+import { RouterLink, RouterLinkActive, Router } from '@angular/router';
+import { AuthService } from '../../auth/auth.service';
+import { AsyncPipe, CommonModule } from '@angular/common';
+import { User } from '../../auth/user.model';
+import { first, Subscription, tap } from 'rxjs';
 
 @Component({
   selector: 'app-nav',
   standalone: true,
   imports: [
+    CommonModule,
     RouterLink,
-    RouterLinkActive
+    RouterLinkActive,
+    AsyncPipe
 ],
   templateUrl: './nav.component.html',
   styleUrl: './nav.component.scss'
 })
-export class NavComponent {
+export class NavComponent implements OnDestroy{
 
+  currentUser: User | null = null;
+  currentUserSubscription: Subscription = this.authService.currentUser$.subscribe(user => {
```

Comment:
As we discussed in the meeting, you can just use the observable directly, instead of subscribing to the value here and assigning it to another variable